### PR TITLE
readme: add 'beginners guide'/library reference with code examples

### PR DESCRIPTION
**Pull request by PettitWesley**
I wrote originally wrote this guide because I need to train one of my coworkers to work on Fluent Bit. I think its generally useful for the project to have a guide like this- diving into the code when you are new is a daunting task.

I think that any engineer with knowledge of C could read the existing code base and figure out how to make changes. However, many would be put off by the initial steep learning curve. I've found with these sorts of things that if you give people a little help in the beginning, it speeds things up a lot. I think it's partly a psychological factor- having a clear starting point and example helps one figure out what they need to learn. Once the path is clear, motivation follows.

### CODE REVIEW

File: Beginners_Guide.md [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+
+    flb_http_client_destroy(client);
+    flb_upstream_conn_release(u_conn);
+    flb_upstream_destroy(upstream);
+
+    return resp;
+}
+```
+
+An `flb_upstream` structure represents a host/endpoint that you want to call. Normally, you'd store this structure somewhere so that it can be re-used. An `flb_upstream_conn` represents a connection to that host for a single HTTP request. The connection structure should not be used for more than one request.
```
Comment (PettitWesley) [inline Beginners_Guide.md]: > The connection structure should not be used for more than one request.

@edsiper this is true even if you use keep alive mode, right?

**Comment by edsiper** [comment]
Thanks! Can u just rename the file to CONTRIBUTING_DEVELOPERS.md or similar?

**Comment by PettitWesley** [comment]
@edsiper Renamed it to DEVELOPER_GUIDE.md

**Comment by edsiper** [comment]
thanks!

**Pull request merged**